### ISSUE TITLE
Resume Interrupted Training on Different Machine

### ISSUE THREAD

**Issue by itsjustrafli** [ultralytics/yolov5#12171]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I've been training yolov5s custom model on a machine, long story short it gets interrupted at 95 epochs out of 185 epochs. I know I can resume the training by raised `--resume` flag, but as far as I know it is the case when you want to resume the training on the same machine.

But in my case, I want to resume the training on a different machine (better resource than the previous machine), how to resume the training on the new machine? Is it just simply raised the `--resume` flag or there is more proper way to resume the training process?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@itsjustrafli hi there,

To resume training on a new machine with better resources, you can simply use the `--resume` flag as you mentioned. This flag will load the most recent checkpoint from the `runs/train` directory and continue training from there.

It's important to ensure that the new machine has the necessary dependencies and datasets set up correctly to ensure a smooth transition. Additionally, if you're using a different number of GPUs or any other hardware changes, you may need to modify the `--batch-size` accordingly.

If you face any issues during the process, please feel free to ask for further assistance.

Thanks and happy training!

**Comment by itsjustrafli** [comment]
Thanks, it is worked!

I have another question related to training, how do I remove "background" class in confusion matrix? I just want my four classes is displayed on my confusion matrix

![confusion_matrix](https://github.com/ultralytics/yolov5/assets/88174587/35241ecd-5e0a-489c-83e4-488ec8b47491)

Thanks in advance!

**Comment by glenn-jocher** [comment]
@itsjustrafli glad to hear that it worked for you!

Regarding your question on removing the "background" class from the confusion matrix, you can achieve this by modifying the `labels` parameter in the `plot_confusion_matrix` function in `utils/plots.py` file.

You will need to change the `labels` list to exclude the "background" class label. For example, if your class labels are ["class1", "class2", "class3", "class4", "background"], you can modify it to ["class1", "class2", "class3", "class4"].

Once you make this change and run the script again, the confusion matrix will only display your four classes without the "background" class.

Let me know if you have any further questions or need any additional assistance.

**Comment by itsjustrafli** [comment]
Thanks for your assistance!

But apparently, I didn't found function named `plot_confusion_matrix` in `utils/plots.py` file. I have search with certain keyword related to confusion matrix, but it returns nothing. Can you please show me the code snippet? I would appreciate it.

Thank you so much for this!

**Comment by glenn-jocher** [comment]
@itsjustrafli hey there!

I apologize for the confusion. The `plot_confusion_matrix` function I mentioned is not available in the current version of the `utils/plots.py` file in the YOLOv5 repository.

To plot the confusion matrix without the "background" class, you can modify the `plot_results()` function in `utils/plots.py`. 

Below is a code snippet that shows how you can accomplish this:

```python
from sklearn.metrics import confusion_matrix
import matplotlib.pyplot as plt

def plot_results():
    ...
    # Generate confusion matrix
    y_true = ... # Ground truth labels
    y_pred = ... # Predicted labels
    labels = ... # Class labels, excluding "background"
    cm = confusion_matrix(y_true, y_pred, labels=labels)

    # Plot confusion matrix
    plt.figure(figsize=(10, 8))
    plt.imshow(cm, interpolation='nearest', cmap=plt.cm.Blues)
    ...
    plt.show()
```

You can modify the `labels` list to include your specific class labels, excluding the "background" class. Running this code will generate the confusion matrix with only your four desired classes.

Once again, I apologize for any confusion caused. Let me know if you have any further questions or need additional help!
